packages/next-app/src/app/pool/[poolId]/(components)/MetadataAttributesTable.tsx

```diff
@@ -111,14 +117,63 @@ function Row({ data }: { data: PoolMetadataAttribute }) {
   );
 }
 
-export function MetadataAttributesTable({ poolId }: { poolId: string }) {
-  const { metadata } = useContext(PoolMetadataContext);
+export function MetadataAttributesTable({ poolId }: { poolId: `0x${string}` }) {
+  const { metadata, handleSetMetadata, handleSetPool } =
+    useContext(PoolMetadataContext);
+
+  handleSetPool(poolId);
+
+  useEffect(() => handleSetMetadata([]), []);
+
+  const { data: pools } = metadataGql.useMetadataPool({
+    poolId,
+  });
+
+  const metadataPool = pools?.pools[0];
+
+  const [poolMetadataCID, setMetadataCID] = useState("");
+  const debouncedCID = useDebounce(poolMetadataCID, 500);
+
+  useEffect(() => {
+    async function fetchData() {
+      const response = await fetch(
+        `https://ipfs.io/ipfs/${metadataPool?.metadataCID}`,
```

Comment:
`#question` I know Piñata has an option to make pins public, but I can't remember whether that's set up by default or not -- have you tested this and are we guaranteeing that the things we've pinned are publicly accessible in this endpoint?

Reply:
[https://ipfs.io/ipfs/QmXJ7yxerhSXQd7DVy2f9CbJ6ieBEGQeiLhDnuRkNL4V8a](https://ipfs.io/ipfs/QmXJ7yxerhSXQd7DVy2f9CbJ6ieBEGQeiLhDnuRkNL4V8a)

My last one is accessible ✅

Reply:
In my tests, the [https://ipfs.io/ipfs/](https://ipfs.io/ipfs/) performed better than https://gateway.pinata.cloud/ipfs/